sat/hms_discovery.py

```diff
@@ -209,19 +230,17 @@ def has_completed(self):
         time we started this waiter.
         """
         try:
-            last_schedule_time = self.hd_cron_job.get_last_schedule_time()
-        except HMSDiscoveryError as err:
-            LOGGER.warning(f'Failed to get last schedule time: {err}')
-            return False
-
-        if not last_schedule_time:
-            LOGGER.debug(f'No record of {self.hd_cron_job.FULL_NAME} being scheduled.')
-            return False
-
-        if last_schedule_time >= self.start_time:
-            return True
-        else:
-            return False
+            jobs = self.hd_cron_job.get_jobs()
+            return any(
+                job.metadata.creation_timestamp > self.start_time
+                for job in jobs
+            )
+        except ApiException as err:
+            raise WaitingFailure(
```

Comment:
This is a slight change in behavior from before, and I just want to make sure we discuss the rationale.

Before, if there was some failure when querying the K8s API for the cronjob's last schedule time, it raised `HMSDiscoveryError` from the K8s `ApiException`, and then we caught it, logged a warning (on line 214 on the left), and then returned False, so it would keep trying until the timeout.

With this change, if there is a failure listing cronjobs through the K8s API, we will get a K8s `ApiException` from `get_jobs`, which is caught, and then we raise a `WaitingFailure` which causes the waiting to stop.

It seems fairly unlikely that we will encounter such an error, but if we do, it seems like it might be beneficial to tolerate it if it's a transient problem. Thoughts?

Reply:
My reasoning was that if the jobs aren't able to be retrieved, then it may be less likely to be a transient error. However, reviewing the previous implementation, the `HMSDiscoveryError` was being raised if the `hms-discovery` cronjob couldn't be read from the K8s API, which is probably the same level of severity. I can revert to continuing to wait, I don't feel strongly about stopping waiting on error.